packages/jest/src/plugins/plugin.ts

```diff
@@ -690,3 +696,26 @@ async function getJestOption<T = any>(
 
   return undefined;
 }
+
+/**
+ * Helper that tries to deduct the name of the CI group, based on the related target name.
+ *
+ * This will work well, when the ci target name follows the documented naming convention or similar (for e.g `test-ci`, `e2e-ci`, `ny-e2e-ci`, etc).
+ *
+ * For example, `test-ci` => `TEST (CI)`,  `e2e-ci` => `E2E (CI)`,  `my-e2e-ci` => `MY E2E (CI)`
+ *
+ *
+ * @param ciTargetName name of the ci target
+ * @returns the deducted group name or `E2E (CI)` if cannot be deducted automatically
+ */
+function deductGroupNameFromTarget(ciTargetName: string | undefined) {
+  if (ciTargetName) {
+    const parts = ciTargetName.split('-').map((v) => v.toUpperCase());
+
+    if (parts.length > 1) {
+      return `${parts.slice(0, -1).join(' ')} (${parts[parts.length - 1]})`;
+    }
+  }
+
+  return 'E2E (CI)'; // default group name when cannot be deducted from target name
+}
```

Comment:
If `ciTargetName` is not defined, we don't need a group name because it won't be inferred. So, the name can always be deducted and the only consideration would be that if there's a single segment, we could append ` (CI)` to it.
```suggestion
function deductGroupNameFromTarget(ciTargetName: string | undefined) {
  if (!ciTargetName) {
    return null;
  }

  const parts = ciTargetName.split('-').map((v) => v.toUpperCase());

  if (parts.length > 1) {
    return `${parts.slice(0, -1).join(' ')} (${parts[parts.length - 1]})`;
  }

  return `${parts[0]} (CI)`; // default group name when there is a single segment
}
```